logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/DeleteMessageUseCase.kt

```diff
@@ -0,0 +1,57 @@
+package com.wire.kalium.logic.feature.message
+
+import com.benasher44.uuid.uuid4
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.logic.sync.SyncManager
+import kotlinx.coroutines.flow.first
+import kotlinx.datetime.Clock
+
+class DeleteMessageUseCase(
+    private val messageRepository: MessageRepository,
+    private val userRepository: UserRepository,
+    private val clientRepository: ClientRepository,
+    private val syncManager: SyncManager,
+    private val messageSender: MessageSender
+) {
+
+    suspend operator fun invoke(conversationId: ConversationId, messageId: String, deleteForEveryOne: Boolean): Either<CoreFailure, Unit> {
+        syncManager.waitForSlowSyncToComplete()
+        val selfUser = userRepository.getSelfUser().first()
+
+        return suspending {
+            val generatedMessageUuid = uuid4().toString()
+
+            clientRepository.currentClientId().flatMap { currentClientId ->
+                val message = Message(
+                    id = generatedMessageUuid,
+                    content = if (deleteForEveryOne)
+                        MessageContent.DeleteMessage(messageId)
+                    else
+                        MessageContent.HideMessage(messageId, conversationId),
+                    conversationId = conversationId,
+                    date = Clock.System.now().toString(),
+                    senderUserId = selfUser.id,
+                    senderClientId = currentClientId,
+                    status = Message.Status.PENDING
+                )
+                messageSender.getRecipientsAndAttemptSend(conversationId, message)
```

Comment:
Might be worth to creating a helper method for constructing messages since only `content` and `conversationId` are relevant the others are always constructed in the same way.

Reply:
Agree, actually thinking of that. but at the moment I'd like to see at least one or two events then we can create a better structure because all of these are kinds of actions/utils/extensions around the Message class.